ext/gmp/gmp.c

```diff
@@ -81,6 +81,16 @@ ZEND_GET_MODULE(gmp)
 static zend_class_entry *gmp_ce;
 static zend_object_handlers gmp_object_handlers;
 
+/* Allocator state: `gmp_request_allocator` is a true TLS global
+ * to indicate whether this thread is currently working on a PHP request.
+ * Just checking EG() is not enough because other modules may be loaded in the webserver. */
+static TSRM_TLS bool gmp_request_allocator = false;
```

Comment:
Couldn't this be a gmp module global?

Reply:
No.

Simplest counter-example:

Imagine you're in an embedded SAPI with multiple threads. If you only run a single PHP interpreter at a time you can safely use NTS in this scenario. When NTS is used, the variable will be a true global, which would break other threads trying to use libgmp.

That aside, AFAIK, a true TLS variable will be faster than having to go through the TSRM too. As other DSO's don't need access to this variable directly there's no reason to put it in the module globals.

Reply:
Ah, thank you!